### PR TITLE
🚀 [Feat:] Download feature

### LINKED ISSUE #101: 🚀Feature: Add a download pixel art button
Currently, the project lacks the ability to download the art created by the user. 
I would like to work on this issue to add a download art functionality to the project.

The download button will be placed below the canvas, and as soon as the user clicks on it. It will download the created art in either JPEG or PNG. Format

Kindly assign this issue to me under the GSSOC'21 program.

Thank you.

### PR DESCRIPTION
**Pull request by amaan-ahmad**
## Related Issue 

- Info about the related issue 

Closes: #101 

### Describe the changes you've made

- I have added a download button at the bottom right corner.
- This button is visible when a canvas is initialized 
- On clicking: It generates `dataURL` from the canvas art and passes that to another utility function called `download` in `app.js`
- `download` function creates an `<a>` tag and embeds the image to its `href` and automatically initiates the download of the image obtained from `dataURL`
- Added class `btn-download` to `style.css` for download button CSS styling.

#### TLDR;

I've just added a download functionality that allows user to download the canvas art they just drew!

Files changed:
- `index.html`
- `app.js`
- `style.css`

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behavior of your code (Write NA if not)
NA

## Additional Info (optional)
This feature downloads the image in PNG format. Open to suggestions from maintainers. ✌

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by adarshvulli** [comment]
Hello ! thanks for the pr! Right now I see the images are downloading in the png format and not in the size of the grid is it possible to add the download option like a jpeg or png. Then download the work in the grid size.

**Comment by amaan-ahmad** [comment]
hey @adarshvulli 
as discussed, yes I'll provide users with an option to choose the downloading format from either JPEG or PNG.

**Comment by amaan-ahmad** [comment]
Hey @adarshvulli ,

I Implemented the download feature for both PNG and JPEG, however, the JPEG image produces a solid black color image no matter canvas has any data or not because the canvas has a transparent background fill and JPEG expects a background fill else it will just produce a black image. Refer to this [stackoverflow solution](https://stackoverflow.com/questions/26742180/canvas-todataurl-results-in-solid-black-image). 
Should I go with it and change the fillStyle of canvas? Asking for your valuable suggestions.  

Thanks, please revert at the earliest.

**Comment by adarshvulli** [comment]
> Hey @adarshvulli ,
> 
> I Implemented the download feature for both PNG and JPEG, however, the JPEG image produces a solid black color image no matter canvas has any data or not because the canvas has a transparent background fill and JPEG expects a background fill else it will just produce a black image. Refer to this [stackoverflow solution](https://stackoverflow.com/questions/26742180/canvas-todataurl-results-in-solid-black-image).
> Should I go with it and change the fillStyle of canvas? Asking for your valuable suggestions.
> 
> Thanks, please revert at the earliest.

Hey, @amaan-ahmad let's proceed with the fillStyle and do you think this will hamper when a user adds his own background color? we need to come up with another alternative at that instance

**Comment by amaan-ahmad** [comment]
@adarshvulli We are only concerned with the background fill being transparent. any color other than transparent whether selected by user or default will not affect the overall application and its purpose.

Because when: 
Background fill = transparent => produces a solid black image for JPEG as it doesn't support a transparent background.
Background fill = any color (other than transparent) => produces the desired result.

Hope this comparison clears many doubts and confusion. Still left with any then let me know ✌

**Comment by adarshvulli** [comment]
> @adarshvulli We are only concerned with the background fill being transparent. any color other than transparent whether selected by user or default will not affect the overall application and its purpose.
> 
> Because when:
> Background fill = transparent => produces a solid black image for JPEG as it doesn't support a transparent background.
> Background fill = any color (other than transparent) => produces the desired result.
> 
> Hope this comparison clears many doubts and confusion. Still left with any then let me know ✌

okay then! Sure! lets go with the fill

**Comment by adarshvulli** [comment]
So ! @amaan-ahmad let's go with the proposed method of the canvas fill.

**Comment by amaan-ahmad** [comment]
Okay @adarshvulli  working on it. ✅

**Review by adarshvulli — CHANGES_REQUESTED** [review]
In app.js let us change  (line number 63  and 64)
 canvas.height = 10 * height;
 canvas.width = 10 * width; 
 this to 
   **canvas.height = 20* height;
    canvas.width = 20 * width;**
    
    cause multiplying with 10 is still a small picture rest all is good and awesome work👌

**Comment by amaan-ahmad** [comment]
Done as suggested @adarshvulli ✌

**Review by adarshvulli — APPROVED** [review]

**Comment by adarshvulli** [comment]
@ankitapuri the pr is good to be merged
